data-service/src/unit-tests/java/uk/gov/gchq/palisade/service/data/service/SimpleDataServiceTest.java

```diff
@@ -18,87 +18,188 @@
+    public static final Context CONTEXT = new Context().purpose(PURPOSE);
+
+    public static final String RULE_MESSAGE = "test-rule";
+
+    public static final AtomicLong RECORDS_RETURNED = new AtomicLong(0);
+    public static final AtomicLong RECORDS_PROCESSED = new AtomicLong(0);
+
+
+    public static class PassThroughRule<T extends Serializable> implements Rule<T> {
+        @Override
+        public T apply(final T record, final User user, final Context context) {
+            return record;
+        }
+    }
+
+    public static final Rules<Serializable> RULES = new Rules<>().addRule(RULE_MESSAGE, new PassThroughRule<>());
+
     // Mocks
     final PersistenceLayer persistenceLayer = Mockito.mock(PersistenceLayer.class);
     final DataReader dataReader = Mockito.mock(DataReader.class);
-    final DataReaderResponse readerResponse = Mockito.mock(DataReaderResponse.class);
     final SimpleDataService simpleDataService = new SimpleDataService(persistenceLayer, dataReader);
-    final OutputStream outputStream = Mockito.mock(OutputStream.class);
+
 
     // Test data
-    final DataRequest dataRequest = DataRequest.Builder.create().withToken("test-request-token").withLeafResourceId("/resource/id");
-    final AuthorisedRequestEntity entity = new AuthorisedRequestEntity(
-            "test-request-token",
-            new User().userId("user-id"),
-            new FileResource().id("/resource/id"),
-            new Context(),
-            new Rules<>()
+    final DataRequest dataRequest = DataRequest.Builder.create().withToken(REQUEST_TOKEN).withLeafResourceId(RESOURCE_ID);
+    final AuthorisedRequestEntity authorisedEntity = new AuthorisedRequestEntity(
+            REQUEST_TOKEN,
+            USER,
+            LEAF_RESOURCE,
+            CONTEXT,
+            RULES
     );
+
     final DataReaderRequest readerRequest = new DataReaderRequest()
-            .user(entity.getUser())
-            .resource(entity.getLeafResource())
-            .context(entity.getContext())
-            .rules(entity.getRules());
+            .user(USER)
+            .resource(LEAF_RESOURCE)
+            .context(CONTEXT)
+            .rules(RULES);
+
+    final AuthorisedDataRequest authorisedDataRequest = AuthorisedDataRequest.Builder.create()
+            .withResource(LEAF_RESOURCE)
+            .withUser(USER)
+            .withContext(CONTEXT)
+            .withRules(RULES);
+
+    final String testResponseMessage = "test response for data request";
 
+    final ResponseWriter responseWriter = new ResponseWriter() {
+        final String testData = testResponseMessage;
+
+        @Override
+        public void close() {
+
+        }
+
+        @Override
+        public ResponseWriter write(final OutputStream outputStream) throws IOException {
+
+            InputStream testInputStream = new ByteArrayInputStream(testData.getBytes());
+            testInputStream.transferTo(outputStream);
+            testInputStream.close();
```

Comment:
Should use "try with resources here".

Reply:
Not sure what you mean.  I need to replace the InputStream that is reading from the resource with an InputStream that will provide a simple text message as the input.  I have done this with use of a variant of the ResponseWriter and a DataReaderResponse fit for this purpose.

Reply:
Ok, now I understand.  I have changed the code over to use this approach.